app/views/partials/TeamNames.scala.html

```diff
@@ -0,0 +1,34 @@
+@*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.cataloguefrontend.connector.model.TeamName
+@import uk.gov.hmrc.cataloguefrontend.config.Constant
+
+@this()
+
+@(
+        thingLinkFunction: (String) => Call,
```

Comment:
It looks like this param exists to support both `repository` and `service` urls. I think using the `repository` endpoint for all cases should be fine, so the reverse route will not need to be passed as a param. (service endpoint only exists for the edge case where the service name doesn't match the repo name, but this is rare, so don't think it's worth over engineering here)

Reply:
I've left it parameterized because the table in each page has columns that go to different pages (repository or service), from a usability perspective we should probably stay consistent rather than have the link go to the repository in the case where other links go to service.

I guess if we refactor the code in the future so that the service page/link doesn't exist anymore then we can probably drop the distinction and the parameter.